src/pages/Overview/TrafficManagement.tsx

```diff
@@ -0,0 +1,288 @@
+import GraphDataSource from 'services/GraphDataSource';
+import {
+  buildGraphAuthorizationPolicy,
+  buildNamespaceInjectionPatch,
+  buildGraphSidecars
+} from 'components/IstioWizards/WizardActions';
+import { AUTHORIZATION_POLICIES } from '../IstioConfigNew/AuthorizationPolicyForm';
+
+type TrafficManagementProps = {
+  opTarget: string;
+  kind: string;
+  isOpen: boolean;
+  nsTarget: string;
+  nsInfo: NamespaceInfo;
+  hideConfirmModal: () => void;
+  load: () => void;
+  duration: DurationInSeconds;
+};
+
+type State = {
+  confirmationModal: boolean;
+  authorizationPolicies: AuthorizationPolicy[];
+  sidecars: Sidecar[];
+  disableOp: boolean;
+  canaryVersion: string;
+};
+
+export default class TrafficManagement extends React.Component<TrafficManagementProps, State> {
+  private promises = new PromisesRegistry();
+  constructor(props: TrafficManagementProps) {
+    super(props);
+    this.state = {
+      confirmationModal: false,
+      authorizationPolicies: [],
+      sidecars: [],
+      disableOp: true,
+      canaryVersion: this.props.kind === 'canary' ? serverConfig.istioCanaryRevision[this.props.opTarget] : ''
+    };
+  }
+
+  componentDidUpdate(prevProps: TrafficManagementProps) {
+    if (prevProps.nsTarget !== this.props.nsTarget || prevProps.opTarget !== this.props.opTarget) {
+      var authorizationPolicies = this.props.nsInfo?.istioConfig?.authorizationPolicies || [];
+      var sidecars = this.props.nsInfo?.istioConfig?.sidecars || [];
+
+      switch (this.props.kind) {
+        case 'injection':
+          this.fetchPermission();
+          break;
+        case 'canary':
+          this.setState({ canaryVersion: serverConfig.istioCanaryRevision[this.props.opTarget] });
+          this.fetchPermission();
+          break;
+        default:
+          if (this.props.opTarget === 'create') {
+            this.generateTrafficPolicies();
+          } else if (this.props.opTarget === 'update') {
+            const remove = ['uid', 'resourceVersion', 'generation', 'creationTimestamp', 'managedFields'];
+            sidecars.map(sdc => remove.map(key => delete sdc.metadata[key]));
+            authorizationPolicies.map(ap => remove.map(key => delete ap.metadata[key]));
+            this.setState({ authorizationPolicies, sidecars });
+          } else if (this.props.opTarget === 'delete') {
+            var nsInfo = this.props.nsInfo.istioConfig;
+            this.setState({
+              authorizationPolicies: nsInfo?.authorizationPolicies || [],
+              sidecars: nsInfo?.sidecars || []
+            });
+            this.fetchPermission();
+          }
+          break;
+      }
+    }
+  }
+
+  fetchPermission = () => {
+    this.promises.register('namespacepermissions', API.getIstioPermissions([this.props.nsTarget])).then(result => {
+      this.setState({
+        confirmationModal: true,
+        disableOp: !result.data[this.props.nsTarget][AUTHORIZATION_POLICIES][this.props.opTarget]
```

Comment:
This sentence won't work here.
The response of the permissions query is similar to:
```
{"authorizationpolicies":{"create":true,"update":true,"delete":true},"gateways":{"create":true,"update":true,"delete":true},"peerauthentications":{"create":true,"update":true,"delete":true},"requestauthentications":{"create":true,"update":true,"delete":true},"serviceentries":{"create":true,"update":true,"delete":true},"sidecars":{"create":true,"update":true,"delete":true}}
```

But you are asking for a "disable/enable" operation.